src/state_root_anchor_service.js

```diff
@@ -0,0 +1,95 @@
+
+const assert = require('assert');
+const interval = require('interval-promise');
+const Mosaic = require('@openstfoundation/mosaic.js');
+
+/** @todo Move into utils. */
+function isObject(obj) {
+  return typeof obj === 'object' && obj !== null && !Array.isArray(obj);
+}
+
+class StateRootAnchorService {
+  constructor(
+    direction,
+    anchorBlockDelay,
+    mosaic,
+  ) {
+    if (direction === 'origin') {
+      this.source = 'origin';
+      this.target = 'auxiliary';
+    } else if (direction === 'auxiliary') {
+      this.source = 'auxiliary';
+      this.target = 'origin';
+    } else {
+      throw new Error(`Unexpected value (${direction}) for direction. `
+        + 'Expected values are: origin, auxiliary.');
+    }
+
+    assert(Number.isInteger(anchorBlockDelay));
+    assert(anchorBlockDelay > 0);
+    this.anchorBlockDelay = anchorBlockDelay;
+
+    assert(isObject(mosaic));
+    this.mosaic = mosaic;
+
+    const anchorContractAddress = this.mosaic[this.target].contractAddresses.Anchor;
+    assert(this.mosaic[this.target].web3.utils.isAddress(anchorContractAddress));
+    this.anchorContractInteract = new Mosaic.ContractInteract.Anchor(
+      this.mosaic[this.target].web3, anchorContractAddress,
+    );
+
+    this.INTERVAL_TIMEOUT = 1000;
+
+    this.run = false;
+  }
+
+  async getAnchorInfo(blockHashOrBlockNumber) {
```

Comment:
probably nicer to return promises instead of making the functions async? Not sure.

Reply:
As function waits for web3.eth.getBlock() to proceed, for me it's readable to have it as async and use await. Otherwise, I should wrap one call into another.

Reply:
OK